lib/mayaUsd/fileio/primUpdaterContext.h

```diff
@@ -67,6 +68,9 @@ class UsdMayaPrimUpdaterContext
     MAYAUSD_CORE_PUBLIC
     MDagPath MapSdfPathToDagPath(const SdfPath& sdfPath) const;
 
+    mutable MayaUsd::ufe::ReplicateExtrasFromUSD _pullExtras;
```

Comment:
These mutable data members feel uncomfortable.  Does the following make any sense:
- initRecursive() writes into the map.  Can we transform it into a static creator method?
- processItem() and finalize() read from the map.  They can be made const methods.
That way, you can pass in the ReplicateExtras objects as const into the context, make them const data members, and provide const accessors to them, keeping the PrimUpdaterContext read-only.

Hope I didn't skip over some crucial detail...

Reply:
Your plan will not fully work because ReplicateExtrasToUSD::processItem writes into the map.

Reply:
O.K., FromUSD works, ToUSD does not :)  Let me think about it some more.

Reply:
Here's a tweak to my suggestion to get it to work: everything as per above, but ReplicateExtrasToUSD::processItem() does a const cast if it needs to write into the map.  Alternately, the map is mutable.  I think this is acceptable, as the fact that processItem() is writing into the map is an invisible implementation detail.  In that way, the prim updater context can appear to be conceptually read-only.  What do you think?